presidio-structured/presidio_structured/analysis_builder.py

```diff
@@ -0,0 +1,163 @@
+from abc import ABC, abstractmethod
+from collections import Counter
+from collections.abc import Iterable
+from typing import Any, Dict, Iterator, Union
+
+from pandas import DataFrame
+from presidio_analyzer import (
+    AnalyzerEngine,
+    BatchAnalyzerEngine,
+    DictAnalyzerResult,
+    RecognizerResult,
+)
+
+from presidio_structured.config import StructuredAnalysis
+
+
+class AnalysisBuilder(ABC):
+    """
+    Abstract base class for a configuration generator.
+    """
+
+    def __init__(self):
+        """Initialize the configuration generator."""
+        self.analyzer = AnalyzerEngine()
+
+    @abstractmethod
+    def generate_analysis(self, data: Union[Dict, DataFrame]) -> StructuredAnalysis:
+        """
+        Abstract method to generate a configuration from the given data.
+
+        :param data: The input data. Can be a dictionary or DataFrame instance.
+        :type data: Union[Dict, DataFrame]
+        :return: The generated configuration.
+        :rtype StructuredAnalysis:
+        """
+        pass
+
+
+class JsonAnalysisBuilder(AnalysisBuilder):
+    """Concrete configuration generator for JSON data."""
+
+    def generate_analysis(self, data: Dict) -> StructuredAnalysis:
+        """
+        Generate a configuration from the given JSON data.
+
+        :param data: The input JSON data.
+        :type data: Dict
+        :return: The generated configuration.
+        :rtype StructuredAnalysis:
+        """
+        batch_analyzer = BatchAnalyzerEngine(analyzer_engine=self.analyzer)
+        analyzer_results = batch_analyzer.analyze_dict(input_dict=data, language="en")
+        return self._generate_analysis_from_results_json(analyzer_results)
+
+    def _generate_analysis_from_results_json(
+        self, analyzer_results: Iterator[DictAnalyzerResult], prefix: str = ""
+    ) -> StructuredAnalysis:
+        """
+        Generate a configuration from the given analyzer results.
+
+        :param analyzer_results: The analyzer results.
+        :type analyzer_results: Iterator[DictAnalyzerResult]
+        :param prefix: The prefix for the configuration keys.
+        :type prefix: str
+        :return: The generated configuration.
+        :rtype StructuredAnalysis:
+        """
+        mappings = {}
+
+        if not isinstance(analyzer_results, Iterable):
+            return mappings
+
+        for result in analyzer_results:
+            current_key = prefix + result.key
+
+            if isinstance(result.value, dict):
+                nested_mappings = self._generate_analysis_from_results_json(
+                    result.recognizer_results, prefix=current_key + "."
+                )
+                mappings.update(nested_mappings.entity_mapping)
+
+            if sum(1 for _ in result.recognizer_results) > 0:
+                for recognizer_result in result.recognizer_results:
+                    mappings[current_key] = recognizer_result.entity_type
```

Comment:
Yes, you are right, this was an oversight/leftover from the first hacky implementation. 

@ebotiab 's changes take the first, I agree makes sense for (semi)structured data, since we should only have one entity type. 

Also added the nuance to the docstring